src/components/show-products/index.js

```diff
@@ -10,16 +11,52 @@ import {
 import { TimeFrames } from '../../utils/timeFrames';
 import { compareDates } from '../../utils/compareDates';
 import ContentContainer from '../content-container';
+import { doc, setDoc, serverTimestamp } from '@firebase/firestore';
+import { db } from '../../lib/firebase';
+
 import './styles.css';
 
 export const ShowProducts = () => {
   const { products, loading } = useProducts();
   const { push } = useHistory();
   const { storedValue } = useLocalStorage(LOCAL_STORAGE_LIST_TOKEN);
   const timeFrames = TimeFrames;
+  const ONE_DAY = 1;
 
-  const handleOnChange = (productID) => {
+  const handleOnChange = (e, productID) => {
     updatePurchaseDate(productID, storedValue);
+    const item = products?.find((product) => product.id === productID);
+    const daysSinceLastTransaction =
+      item.createdAt.toDate() !== 0 ? compareDates(item.createdAt.toDate()) : 0;
```

Comment:
Only ideas:

Lines 28-29 could be moved to external functions like `findProductById` and `getDaysSinceLastTransaction`

For me, it could be clearer, easy to read and reuse those functions in other places. 😄